reana_server/rest/workflows.py

```diff
@@ -1190,7 +1222,7 @@ def upload_file(workflow_id_or_name, user):  # noqa
     except KeyError as e:
         logging.error(traceback.format_exc())
         return jsonify({"message": str(e)}), 400
-    except ValueError as e:
+    except (REANAQuotaExceededError, ValueError) as e:
```

Comment:
Works fine :+1:

One observation though: when using `reana-client==0.7.1` the transaction appears as if it were OK, but the file is not uploaded.

```console
$ # with reana-client on master:
$ reana-client upload -w demo README.rst
Something went wrong while uploading .../README.rst:
User quota exceeded.

$ # with reana-client==0.7.1
$ reana-client upload -w demo README.rst
File /README.rst was successfully uploaded.
```

Can we do something so that older clients wouldn't appear as if the transaction succeeded? We can do it as part of another issue.

Reply:
Ah, I recall this problem. I think it was fixed [here](https://github.com/reanahub/reana-client/commit/d2a5f2ffe83f5795da9e0cdcb3657d2157893a61), maybe this commit should have gone to `maint-0.7`... Should we move it?

Reply:
I confirm it was the case. I've created another PR pointing to r-client `maint-0.7` to fix this: https://github.com/reanahub/reana-client/pull/445